ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatRecord.java

```diff
@@ -657,7 +658,88 @@ public static void serializeMetadataContentRecord(ByteBuffer outputBuffer, List<
         StoreKey storeKey = storeKeyFactory.getStoreKey(stream);
         keys.add(storeKey);
       }
-      return keys;
+      return new MultiPartMetadata(keys);
+    }
+  }
+
+  /**
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   * |         |              |              |            |            |          |
+   * | version |  chunk size  |  total size  |    key1    |     key2   |  ......  |
+   * |(2 bytes)|  (4 bytes)   |  (8 bytes)   |            |            |  ......  |
+   * |         |              |              |            |            |          |
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   *  version         - The version of the metadata content record
+   *
+   *  max chunk size  - maximum size of a data chunk in bytes
+   *
+   *  total size      - total size of the object this metadata describes.
+   *
+   *  key1            - first key to be part of metadata blob
+   *
+   *  key2            - second key to be part of metadata blob
+   *
+   */
+  public static class Metadata_Content_Format_V2 {
+    public static final int Max_Chunk_Size_Field_Size_In_Bytes = 4;
+    public static final int Total_Size_Field_Size_In_Bytes = 8;
```

Comment:
I'm not sure.  Maybe it was in case someone wanted to use some custom deserialization logic and needed this information.  I'm fine with making these private, though, because I don't see much of a use for them outside this class.